To: vim_dev@googlegroups.com
Subject: Patch 7.3.192
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.192
Problem:    Ex command ":s/ \?/ /g" splits multi-byte characters into bytes.
	    (Dominique Pelle)
Solution:   Advance over whole character instead of one byte.
Files:	    src/ex_cmds.c


*** ../vim-7.3.191/src/ex_cmds.c	2011-05-10 16:41:13.000000000 +0200
--- src/ex_cmds.c	2011-05-19 14:23:33.000000000 +0200
***************
*** 4625,4631 ****
  			 * for a match in this line again. */
  			skip_match = TRUE;
  		    else
! 			++matchcol; /* search for a match at next column */
  		    goto skip;
  		}
  
--- 4625,4639 ----
  			 * for a match in this line again. */
  			skip_match = TRUE;
  		    else
! 		    {
! 			 /* search for a match at next column */
! #ifdef FEAT_MBYTE
! 			if (has_mbyte)
! 			    matchcol += mb_ptr2len(sub_firstline + matchcol);
! 			else
! #endif
! 			    ++matchcol;
! 		    }
  		    goto skip;
  		}
  
*** ../vim-7.3.191/src/version.c	2011-05-19 13:40:47.000000000 +0200
--- src/version.c	2011-05-19 14:28:44.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     192,
  /**/

-- 
From "know your smileys":
 :-F	Bucktoothed vampire with one tooth missing

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
